### DISCUSSION TITLE
Cannot change package of a file

### DISCUSSION THREAD

**Discussion by Samar-exe** [mfussenegger/nvim-jdtls#609] [Q&A]
I've setup my configuration correctly since jdtls is doing it's thing,  auto completion is working properly too, but the error arises when I make a new file using touch or from Tree. Once a file is made when I declare a package it give me the error "The declared package does not match the expected package ' ' ". I tried -  First making a text file then declaring the package then renaming the file to .java. I've searched the solution on Google tying to solve this myself, i encountered an answer on StackOverflow stating that I need to upgrade my source path to my package, but I don't know how to do it. Any help would be appreciated.

In the image I made the file with .txt first then declared the package name to src then converted the file to .java, after that when I tried to change the package name to Samar it's giving me this error
![IMG20240122093456.jpg](https://github.com/mfussenegger/nvim-jdtls/assets/91871522/019fcdd4-ad9f-440e-8192-d177488b1903)

**Comment by mfussenegger** [comment]
In java the directory layout needs to match the package definition.

Usually you have `src/main/java` where `main` indicates the source set. That's usually either main or test. `java` indicates the language, that's useful for projects with use more than one JVM language. Say kotlin and java, or java and scala. After that you've a folder matching each part of the package.

So if you have a file `FibonacciNumber` with a package declaration `samar`, you must have a `src/main/java/samar/FibonacciNumber.java`
